mixer/pkg/pool/goroutine.go

```diff
@@ -21,17 +21,27 @@ import (
 // WorkFunc represents a function to invoke from a worker.
 type WorkFunc func()
 
+// WorkFuncWithParam represents a function to invoke from a worker. The parameter is passed on the side
+// to avoid creating closures and allocating.
+type WorkFuncWithParam func(param interface{})
```

Comment:
If we want to discourage using closures, we should just remove the other option and only keep 
`type WorkFunc func(param interface{})`
it is easy change the existing code to do this.

Reply:
I tried going through the existing code and updating it to use non-closure generating one, but the existing call-sites will still cause similar allocations with this scheme and it will be a wash. This approach yields benefits when the closed-over-state is already on the heap and can be pointed by a single pointer.

Still I updated this to be the main method of use, and updated the call sites to pass a nil parameter.